src/bootloader/rollbacks/fiovb.h

```diff
@@ -35,9 +35,21 @@ class FiovbRollback : public Rollback {
   }
 
   void installNotify(const Uptane::Target& target) {
+    std::string version = getVersion(target);
+    if (version.empty()) {
+      return;
+    }
     std::string sink;
-    if (Utils::shell("fiovb_setenv bootupgrade_available 1", &sink) != 0) {
-      LOG_WARNING << "Failed to set bootupgrade_available";
+    if (Utils::shell("fiovb_printenv bootfirmware_version", &sink) != 0) {
+      LOG_WARNING << "Failed to read bootfirmware_version";
+      return;
```

Comment:
@igoropaniuk same here

Reply:
Here it's different. We have to consider the case when `bootfirmware_version` is not set yet (for example, just after flashing via uuu)

@ricardosalveti could you please provide your input here

Reply:
Yes, if not yet set we can set to the current version.